social-api/src/main/java/org/geekbeacon/social/service/impl/InstagramSocialDataServiceImpl.java

```diff
@@ -53,13 +68,91 @@ public int getCount() {
         }
     }
 
+    /**
+     * @see SocialDataService#authorizeApplication()
+     */
     @Override
-    public AuthResponse authorize() {
+    public AuthResponse authorizeApplication() {
         throw new UnsupportedOperationException("Not yet implemented");
     }
 
+    /**
+     * @see SocialDataService#saveRequestToken(String, String)
+     */
     @Override
     public void saveRequestToken(String token, String verifier) {
         throw new UnsupportedOperationException("Not yet implemented");
     }
+
+    /**
+     * @see SocialDataService#getSocialActivity(int)
+     */
+    @Override
+    public List<SocialActivity> getSocialActivity(int limit) {
+        List<SocialActivity> activities = Lists.newArrayList();
+        try {
+            HttpUriRequest request = getRequest();
+            HttpResponse response = client.execute(request);
+            validateRequest(response, true);
+            String rawJson = EntityUtils.toString(response.getEntity());
+            Object document = Configuration.defaultConfiguration().jsonProvider().parse(rawJson);
+
+            //get Array of all posts
+            JSONArray array = safeExtract(document, "$.graphql.user.edge_owner_to_timeline_media.edges[*].node", new JSONArray());
+
+            for (Object item : array) {
+                if (item instanceof Map) {
+                    SocialActivity.SocialActivityBuilder builder = SocialActivity.builder();
+                    Map<String, Object> items = (Map<String, Object>) item;
+                    //Convert items back to String to easily extract using JsonPath
+                    String jsonString = mapper.writeValueAsString(items);
+                    Object nodeDocument = Configuration.defaultConfiguration().jsonProvider().parse(jsonString);
+                    String caption = safeExtract(nodeDocument, "$.edge_media_to_caption.edges[0].node.text", "");
+                    String shortCode =items.get("shortcode").toString();
+                    if(StringUtils.isNotEmpty(shortCode)) {
+                       String originalUri = String.format(INSTAGRAM_URL_FORMAT, shortCode);
+                       builder.sourceUri(new URI(originalUri));
+                    }
+                    Map<String, Object> socialMediaLove = Maps.newHashMap();
+                    builder.socialLove(socialMediaLove);
+
+                    int count =  safeExtract(nodeDocument, "$.edge_media_to_comment.count", 0);
+                    socialMediaLove.put("comments", count);
+                    count = safeExtract(nodeDocument, "$.edge_liked_by.count", 0);
+                    socialMediaLove.put("likes", count);
+
+
+                    builder.text(caption);
+                    builder.mediaUrl(items.getOrDefault("display_url", "").toString());
+                    builder.isVideo(items.get("is_video") == null ? false : (Boolean) items.get("is_video"));
```

Comment:
No, the null check is valid, since i don't want  the tri-state that Java loves (true/false/null).  I can  use booleanUtils  instead with contains when  reading it to avoid that.